### PR TITLE
Update AllColumnNamesValueReader.cs

### PR DESCRIPTION
**Pull request by ratminer**
Columns with empty names are added to ExcelHeading.ColumnNames, but not added to the ExcelHeading.NameMapping
this results in the following exception being thrown when attempting to map to a Dictionary after ExcelHeading.GetColumnIndex is called

ExcelMapper.ExcelMappingException: Column "" does not exist in ["header1", "header2", "header3"]
   at ExcelMapper.ExcelHeading.GetColumnIndex(String columnName)
   at ExcelMapper.Readers.AllColumnNamesValueReader.<>c__DisplayClass0_0.<TryGetValues>b__0(String columnName)
   at System.Linq.Enumerable.SelectArrayIterator`2.MoveNext()
   at ExcelMapper.ManyToOneDictionaryMap`1.TryGetValue(ExcelSheet sheet, Int32 rowIndex, IExcelDataReader reader, MemberInfo member, Object& value)
   at ExcelMapper.ExcelClassMap.TryGetValue(ExcelSheet sheet, Int32 rowIndex, IExcelDataReader reader, MemberInfo member, Object& result)
   at ExcelMapper.ExcelSheet.TryReadRow[T](T& value)

### CODE REVIEW

**Comment by hughbe** [comment]
Could you share a sample excel? Would be good to see if we can add a test

File: tests/ExcelMapper/ExcelSheetTests.cs [hunk 1]
```diff
@@ -519,5 +519,31 @@ private class BlankLinesClass
...
+
+                mapping.Map(t => t.RawRow);
+                importer.Configuration.RegisterClassMap(mapping);
+                ExcelSheet sheet = importer.ReadSheet();
+
+                sheet.ReadRows<Dictionary>().ToArray();
+            }
+            catch (Exception e)
+            {
+                Assert.False(true, e.Message);
```
Comment (hughbe) [inline tests/ExcelMapper/ExcelSheetTests.cs:545]: You can remove this whole try catch

File: tests/ExcelMapper/ExcelSheetTests.cs [hunk 2]
```diff
@@ -519,5 +519,31 @@ private class BlankLinesClass
...
+            try
+            {
+                using var importer = Helpers.GetImporter("ManyToOneMappingIssue.xlsx");
+                var mapping = new ExcelClassMap<Dictionary>();
+
+                mapping.Map(t => t.RawRow);
+                importer.Configuration.RegisterClassMap(mapping);
+                ExcelSheet sheet = importer.ReadSheet();
+
+                sheet.ReadRows<Dictionary>().ToArray();
```
Comment (hughbe) [inline tests/ExcelMapper/ExcelSheetTests.cs:541]: Ideally let’s test the actual contents of the rows (first row should be enough), e.g. `Assert.Equal(??, rows.Count)` and `Assert.Equal(??, rows[0].Count)`

**Review by hughbe — CHANGES_REQUESTED** [review]
Thanks for the tests!

**Comment by ratminer** [comment]
I've added a test file

It seems the issue was a cell in the header row after the table data that had a bold text formatting

File: tests/ExcelMapper/ExcelSheetTests.cs (hunk 2 above)
Comment (ratminer) [reply tests/ExcelMapper/ExcelSheetTests.cs]: updated with asserts for # of rows, and # columns in first row

**Comment by hughbe** [comment]
Thanks so much!

**Pull request merged**